create_aio_app/main.py

```diff
@@ -11,16 +12,23 @@ def main():
     args = parse_arguments()
     template_path = str(parent / 'template')
 
-    if not args.get('name'):
-        result = cookiecutter(template_path)
-    else:
-        ctx = {
-            'project_name': args.get('name'),
-            'use_postgres': 'n' if args.get('without_postgres') else 'y',
-            'use_redis': 'y' if args.get('redis') else 'n',
+    kwargs = {}
+
+    if args.get('name'):
+        kwargs = {
+            'no_input': True,
+            'extra_context': {
+                'project_name': args.get('name'),
+                'use_postgres': 'n' if args.get('without_postgres') else 'y',
+                'use_redis': 'y' if args.get('redis') else 'n',
+            },
         }
 
-        result = cookiecutter(template_path, no_input=True, extra_context=ctx)
+    try:
+        result = cookiecutter(template_path, **kwargs)
+    except OutputDirExistsException as exc:
+        print('\n\n\033[91mDirectory with such name already exists!\033[00m\n')
+        return
```

Comment:
why do you use `return` here?

Reply:
Because otherwise it'll probably corrupt data which might've been inside of pre-existing directory.

Reply:
@Arfey to exit function earlier.

Reply:
👍